.github/workflows/ci.yml

```diff
@@ -51,10 +77,27 @@ jobs:
     runs-on: ubuntu-latest
     timeout-minutes: 10
     steps:
+      - uses: actions/checkout@v4
+        with:
+          persist-credentials: false
+      - name: Get Cache Key
+        id: get-cache-key
+        run: |
+          echo "key=$(git ls-tree -r HEAD | grep '^160000' | sha256sum | cut -d " " -f 1)" >> $GITHUB_OUTPUT
+      - uses: actions/cache@v4
+        with:
+          path: .git/modules
+          key: submodule-${{ steps.get-cache-key.outputs.key }}
+          restore-keys: |
+            submodule-
       - uses: actions/checkout@v4
         with:
           persist-credentials: false
           submodules: true
+      - name: Sparse Checkout
+        run: |
+          $(cd libs/gi/dm/GenshinData  && git sparse-checkout set TextMap ExcelBinOutput)
```

Comment:
Can we do something similar for our local checkouts? I did find it annoying we are checking out the entire repo just for some excels and strings.

Reply:
Doesn't have to be this PR ofc

Reply:
Yes, but I don't think its something that we could/should check into the repo.
You can run these commands locally to modify your local checkout